### DISCUSSION TITLE
Argdown Support

### DISCUSSION THREAD

**Discussion by aadityabhatia** [quarto-dev/quarto-cli#8789] [Feature Requests]
### Description

[Argdown](https://argdown.org/) is a popular markdown-compatible syntax for complex argumentation that is especially relevant in the field of AI argumentation. It would be nice to have built in support for argdown. It already comes with a [pandoc filter](https://argdown.org/guide/publishing-argdown-markdown-with-pandoc.html), so supporting it in quarto should be fairly trivial.

**Comment by aadityabhatia** [comment]
Additionally, it would help to describe more clearly how to use arbitrary pandoc filters with quarto in the [filter documentation](https://quarto.org/docs/extensions/filters.html). Figured it out after searching through the issues - #5392 in particular explained that arbitrary binaries are no longer automatically resolved.

To get argdown filter to work with quarto:

```
npm install -g @argdown/pandoc-filter @argdown/image-export                                                                                                                                   
which argdown-filter
```

```yml
filters:
  - /usr/local/bin/argdown-filter # replace this with resolved executable path
```

**Reply by cderv** [reply]
Did you try package the filter as a Quarto extension: https://quarto.org/docs/extensions/filters.html ? 

This is an interesting use case as this filter is available through npm initially. 🤔 

This could work, but not perfectly yet maybe. It would be interesting to have feedback on that if someone tries it.

**Reply by aadityabhatia** [reply]
I have very no experience with Quarto extensions and the documentation seems to favor lua. I am not sure how to package a JS-based filter available from npm as a plugin. Would love to give it a try if I could see an example.

**Reply by mcanouil** [reply]
Take a look at the other extensions.

**Comment by mcanouil** [comment]
What would be the added value to add a dependency to Quarto compared to the use of the filter (via extension)?

**Reply by aadityabhatia** [reply]
I think using filter/extension would suffice. However, Argdown is a very simple and powerful format with a stable API, so being able to support it natively will make it more discoverable and accessible for those working in the argumentation domain.

**Reply by cscheid** [reply]
Let me be frank, we have other things on our goals that we consider to be more important for the larger community of Quarto users. With that said, pull requests and extensions are always welcome.
